app/script/media/MediaStreamHandler.js

```diff
@@ -268,7 +268,8 @@ z.media.MediaStreamHandler = class MediaStreamHandler {
     return constraintsPromise
       .then(streamConstraints => this.requestMediaStream(mediaType, streamConstraints))
       .then(mediaStreamInfo => {
-        if (!this.mediaStreamInUse()) {
+        if (!isPreferenceChange && !this.mediaStreamInUse()) {
```

Comment:
I still believe we just adjust this condition as we would fix a bunch of additional scenarios. Currently the stream will be erroneously released if you are in a video call, open the preferences, and then end the call from the remote side.